e2store/src/era2.rs

```diff
@@ -123,22 +120,60 @@ impl Era2 {
 
                 self.pending_storage_entries -= 1;
                 let entry: Entry = storage.clone().try_into()?;
-                self.e2store_stream.append_entry(&entry)?;
+                self.writer.append_entry(&entry)?;
                 entry.value.len()
             }
         };
         Ok(size)
     }
+
+    pub fn flush(&mut self) -> anyhow::Result<()> {
+        self.writer.flush()
+    }
+}
+
+/// The `Era2` streaming reader.
+///
+/// Unlike [crate::era::Era] and [crate::era1::Era1], the `Era2` files are too big to be held in
+/// memory.
+pub struct Era2Reader {
+    pub version: VersionEntry,
+    pub header: HeaderEntry,
+
+    reader: E2StoreStreamReader,
+    pending_storage_entries: u32,
+    path: PathBuf,
+}
+
+impl Era2Reader {
+    pub fn new(path: &Path) -> anyhow::Result<Self> {
```

Comment:
I think we should still do open instead of new, because we are opening an era2 file